plugins/pkg/file/fs_test.go

```diff
@@ -16,28 +16,74 @@ package file
 
 import (
 	"os"
+	"path/filepath"
+	"sync"
 	"testing"
-	"time"
 
+	"github.com/fsnotify/fsnotify"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestFileMtimeDetection(t *testing.T) {
-	defaultFs = newFS(2000 * time.Millisecond)
+func TestFileIsChanged(t *testing.T) {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	i := 4
 
-	tmpfile, _ := os.CreateTemp("", "example")
-	defer os.Remove(tmpfile.Name()) // clean up
+	tmpfile, _ := os.CreateTemp("./", "example")
+	defer func(name string) {
+		err := os.Remove(name)
+		if err != nil {
+			t.Logf("%v", err)
+		}
+	}(tmpfile.Name())
+	file, err := Stat(tmpfile.Name())
 
-	f, err := Stat(tmpfile.Name())
+	assert.NoError(t, err)
+	assert.Equal(t, tmpfile.Name(), file.Name)
+	err = WatchFiles(func() {
+		wg.Add(1)
+		defer wg.Done()
+		mu.Lock()
+		i = 5
+		mu.Unlock()
+	}, file)
 	assert.Nil(t, err)
-	assert.False(t, IsChanged(f))
-	time.Sleep(1000 * time.Millisecond)
 	tmpfile.Write([]byte("bls"))
-	tmpfile.Close()
-	assert.False(t, IsChanged(f))
+	tmpfile.Sync()
+	wg.Wait()
+
+	err = WatchFiles(func() {}, nil)
+	assert.Error(t, err, "file pointer cannot be nil")
+
+	mu.Lock()
+	assert.Equal(t, 5, i)
+	mu.Unlock()
+
+	watcher, err := fsnotify.NewWatcher()
+	assert.NoError(t, err)
+	defer watcher.Close()
+	fs := &Fsnotify{
+		WatchedFiles: make(map[string]struct{}),
+		Watcher:      watcher,
+	}
+	tmpfile, err = os.CreateTemp("/tmp", "test")
+	assert.Nil(t, err)
+	defer os.Remove(tmpfile.Name())
+	tmpDir := filepath.Dir(tmpfile.Name())
+	fs.WatchedFiles[tmpDir] = struct{}{}
+	err = fs.AddFiles(tmpDir)
+	assert.NoError(t, err)
+
+	onChangeCalled := false
+	onChange := func() {
+		onChangeCalled = true
+	}
+
+	go fs.watchFiles(onChange, fs.Watcher, tmpDir)
+
+	_, exists := fs.WatchedFiles[tmpDir]
+
+	assert.True(t, exists)
```

Comment:
What does the assert check for? Remove `go fs.watchFiles(onChange, fs.Watcher, tmpDir)`, it still passes. We should check the behavior of `fs.watchFiles`